api/src/main/java/org/motechproject/nms/api/web/KilkariController.java

```diff
@@ -74,14 +80,27 @@ public InboxResponse getInboxDetails(@RequestParam(required = false) Long callin
             throw new NotFoundException(String.format(NOT_FOUND, "callingNumber"));
         }
 
-        Set<Subscription> subscriptions = subscriber.getActiveSubscriptions();
+        Set<Subscription> subscriptions = subscriber.getAllSubscriptions();
```

Comment:
Yep -- once you read InboxServiceImpl.getInboxMessage you'll see how it works. We need to return the final message in a campaign for 7 days after a subscription has completed. After that we return no inbox.